eng/common/pipelines/templates/jobs/archetype-spec-gen-sdk.yml

```diff
@@ -0,0 +1,201 @@
+    displayName: 'Path to API specification file (TypeSpec/OpenAPI)'
+  - name: SkipPullRequestCreation
+    type: boolean
+    default: false
+
+jobs:
+- job:
+  displayName: 'SDK Generation'
+
+  variables:
+    - template: /eng/pipelines/templates/variables/image.yml
+    - name: NodeVersion
+      value: '22.13.x'
+    - name: PythonVersion
+      value: '3.13'
+    - name: SpecGenSdkVersion
+      value: 'latest'
+    - name: SdkArtifactName
+      value: SDK_Artifact
+
+  pool:
+    name: $(LINUXPOOL)
+    vmImage: $(LINUXVMIMAGE)
+
+  steps:
+    - checkout: none
+
+    - pwsh: |
+        $urlPattern = '^https://github\.com/(?<organization>[^/]+)/(?<repository>[^/]+)'
+        if ('${{ parameters.SpecRepoUrl }}' -match $urlPattern) {
+          $specRepoOwner = $Matches['organization']
+          Write-Host "##vso[task.setvariable variable=SpecRepoOwner]$specRepoOwner"
+          Write-Host "SpecRepoOwner variable set to: $specRepoOwner"
+
+          $specRepoName = $Matches['repository']
+          Write-Host "##vso[task.setvariable variable=SpecRepoName]$specRepoName"
+          Write-Host "SpecRepoName variable set to: $specRepoName"
+
+          $specRepoDirectory = "$(System.DefaultWorkingDirectory)/$specRepoName"
+          Write-Host "##vso[task.setvariable variable=SpecRepoDirectory]$specRepoDirectory"
+          Write-Host "SpecRepoDirectory variable set to: $specRepoDirectory"
+        }
+
+        if ('${{ parameters.SdkRepoUrl }}' -match $urlPattern) {
+          if ('${{ parameters.SpecRepoUrl }}'.EndsWith('-pr') -and (-not '${{ parameters.SdkRepoUrl }}'.EndsWith('-pr'))) {
+            Write-Host "SdkRepoUrl must be a private repository if SpecRepoUrl is a private repository."
+            Exit 1
+          }
+
+          if ('$(SdkRepoOwner)' -eq '') {
+            $sdkRepoOwner = $Matches['organization']
+            Write-Host "##vso[task.setvariable variable=SdkRepoOwner]$sdkRepoOwner"
+            Write-Host "SdkRepoOwner variable set to: $sdkRepoOwner"
+          }
+
+          $sdkRepoName = $Matches['repository']
+          Write-Host "##vso[task.setvariable variable=SdkRepoName]$sdkRepoName"
+          Write-Host "SdkRepoName variable set to: $sdkRepoName"
+
+          $sdkRepoDirectory = "$(System.DefaultWorkingDirectory)/$sdkRepoName"
+          Write-Host "##vso[task.setvariable variable=SdkRepoDirectory]$sdkRepoDirectory"
+          Write-Host "SdkRepoDirectory variable set to: $sdkRepoDirectory"
+        }
+
+        if ([string]::IsNullOrEmpty($SpecRepoOwner) -or [string]::IsNullOrEmpty($SpecRepoName) -or [string]::IsNullOrEmpty($SdkRepoOwner) -or [string]::IsNullOrEmpty($SdkRepoName)) {
+          Write-Host "One or more required variables are null or empty. Ensure that SpecRepourl and SdkRepoUrl are set to valid GitHub repository URLs."
+          Exit 1
+        }
+
+        $sdkRepoNameForSpecGen = $sdkRepoName
+        if ($sdkRepoName.EndsWith('-for-net')) {
```

Comment:
@raych1 is this something we can add to the backlog to clean-up? I would like to eliminate the "-track2" from the name if possible.

Reply:
It's in the backlog, we will clean this up while rolling out this pipeline for spec PR scenario. See https://github.com/Azure/azure-sdk-tools/pull/9577#discussion_r1919030709 for details.